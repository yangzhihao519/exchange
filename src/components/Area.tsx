import React from 'react';
import Money from '../models/money'
import Currency from '../models/currency'
import ChangeExchangeValueField from '../components/ChangeExchangeValueField'
import CurrencyList from '../components/CurrencyList'

interface AreaProps {
    allCurrencies: Currency[],
    initBalance: Money,
    exchange: Money,
    isSellingOut: boolean
    onInputChange: (value: number, isSellingOut: boolean) => void
    onSelectCurrency: (selectedCurrencyCode: string, isSellingOut: boolean) => void
}

interface AreaState {}

class Area extends React.Component<AreaProps, AreaState>{
    render() {
        let { allCurrencies, initBalance,exchange, isSellingOut, onInputChange, onSelectCurrency } = this.props;
        let currentCurrency = allCurrencies.find((item) => {return item.code === exchange.currencyCode})

        return(
            <div className={isSellingOut? "area area--sellout" : "area area--buyin"}>
                <div className='area__pocket'>
                    <CurrencyList allCurrencies={allCurrencies} 
                                    onSelectCurrency={onSelectCurrency} 
                                    selectedCurrencyCode={exchange.currencyCode}
                                    isSellingOut={isSellingOut}/>
                    <div className="area__balance">Balance: {currentCurrency && currentCurrency.symbol}{initBalance.value}</div>
                </div>
                <div className='area__input'>
                    <ChangeExchangeValueField isSellingOut={isSellingOut} exchange={exchange} onInputChange={onInputChange}/>
                </div>
            </div>
        )
    }
}
  
export default Area;