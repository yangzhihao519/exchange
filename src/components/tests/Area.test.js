import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Area from '../Area';

const mockFunction = jest.fn();

describe('Area', () => {
  it('renders without crashing given the required props', () => {
    const props = {
        allCurrencies: [],
        initBalance: {},
        exchange: {},
        isSellingOut: false,
        onInputChange: mockFunction,
        onSelectCurrency: mockFunction
    }
    const wrapper = shallow(<Area {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})