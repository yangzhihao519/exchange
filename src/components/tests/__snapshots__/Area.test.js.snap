// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area renders without crashing given the required props 1`] = `
<div
  className="area area--buyin"
>
  <div
    className="area__pocket"
  >
    <CurrencyList
      allCurrencies={Array []}
      isSellingOut={false}
      onSelectCurrency={[MockFunction]}
    />
    <div
      className="area__balance"
    >
      Balance: 
    </div>
  </div>
  <div
    className="area__input"
  >
    <ChangeExchangeValueField
      exchange={Object {}}
      isSellingOut={false}
      onInputChange={[MockFunction]}
    />
  </div>
</div>
`;
